### PR TITLE
Support CMAKE_SYSROOT in CMakeToolchain

### PR DESCRIPTION
**Pull request by czoido**
Changelog: Feature: Add `CMAKE_SYSROOT` support for `CMakeToolchain`.
Changelog: Feature: Add `--sysroot` support for `AutotoolsToolchain` and remove support for `cpp_info.sysroot` in `AutotoolsDeps`.
Changelog: Feature: Add `tools.build:sysroot` conf.

Docs: https://github.com/conan-io/docs/pull/2564

### CODE REVIEW

File: conan/tools/cmake/toolchain/blocks.py [hunk 1]
```diff
@@ -824,14 +828,18 @@ def context(self):
 
         system_name, system_version, system_processor = self._get_cross_build()
 
+        cmake_sysroot = self._conanfile.conf.get("tools.cmake.cmaketoolchain:cmake_sysroot")
```
Comment (memsharded) [inline conan/tools/cmake/toolchain/blocks.py]: We have ``cpp_info.sysroot``, aren't we processing it anymore?

Comment (memsharded) [reply conan/tools/cmake/toolchain/blocks.py]: Ok, yes, lets use the ``conf`` and forget about ``cpp_info.sysroot``, sounds good @lasote?

**Review by memsharded — APPROVED** [review]

File: conan/tools/cmake/toolchain/blocks.py (hunk 1 above)
Comment (memsharded) [reply conan/tools/cmake/toolchain/blocks.py]: The only thing, maybe better generalize the name to something like ``tools.buid:sysroot``, so it makes sense for autotools and other build systems.

**Review by memsharded — CHANGES_REQUESTED** [review]

**Comment by czoido** [comment]
> The only thing, maybe better generalize the name to something like tools.buid:sysroot, so it makes sense for autotools and other build systems.

Yes, I'll check the autotools case as well and make something generic if needed 👍

File: conan/tools/cmake/toolchain/blocks.py [hunk 2]
```diff
@@ -824,14 +828,21 @@ def context(self):
 
         system_name, system_version, system_processor = self._get_cross_build()
 
+        # This is handled by the tools.apple:sdk_path and CMAKE_OSX_SYSROOT in Apple
+        cmake_sysroot = None
+        if not is_apple_os(self._conanfile.settings.get_safe("os")):
```
Comment (memsharded) [inline conan/tools/cmake/toolchain/blocks.py]: why? As this is a defined conf, is it really necessary to not apply it the OS is an apple one?
Is it possible that there there are some scenario (like cross-building or the like) that would want to use the ``sysroot``? Wouldn't it be enough that the user don't define ``tools.build:sysroot`` in their profile (or inside the package, the package recipe would know)

File: conan/tools/gnu/autotoolstoolchain.py [hunk 3]
```diff
@@ -64,7 +66,12 @@ def __init__(self, conanfile, namespace=None):
                 # https://man.archlinux.org/man/clang.1.en#Target_Selection_Options
                 self.apple_arch_flag = "-arch {}".format(apple_arch) if apple_arch else None
                 # -isysroot makes all includes for your library relative to the build directory
-                self.apple_isysroot_flag = "-isysroot {}".format(sdk_path) if sdk_path else None
+                self.sysroot_flag = "-isysroot {}".format(sdk_path) if sdk_path else None
+
+        if not is_apple_os(self._conanfile.settings.get_safe("os")):
```
Comment (memsharded) [inline conan/tools/gnu/autotoolstoolchain.py]: Same as above?

**Review by memsharded — APPROVED** [review]
Looking good

File: conan/tools/cmake/toolchain/blocks.py (hunk 2 above)
Comment (czoido) [reply conan/tools/cmake/toolchain/blocks.py]: Yes, maybe it's better not to do this, I'll remove the check 👍

**Pull request merged**